File: leetcode/editor/en/[1768]Merge Strings Alternately.java

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public String mergeAlternately(String word1, String word2) {
        char[] w1 = word1.toCharArray();
        char[] w2 = word2.toCharArray();
        StringBuilder sb = new StringBuilder();
        int i = 0;
        int j = 0;
        while (i < w1.length || j < w2.length) {
            if (i < w1.length) {
                sb.append(w1[i]);
                i++;
            }
            if (j < w2.length) {
                sb.append(w2[j]);
                j++;
            }
        }
        return sb.toString();
    }
}
//leetcode submit region end(Prohibit modification and deletion)

File: leetcode/editor/en/[1768]Merge Strings Alternately.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function (word1, word2) {
    let res = "";
    let i = 0;
    let j = 0;
    while (i < word1.length && j < word2.length) {
        res += word1.charAt(i++);
        res += word2.charAt(j++);
    }
    if (i < word1.length) {
        res += word1.substring(i);
    }
    if (j < word2.length) {
        res += word2.substring(j);
    }
    return res;
};
//leetcode submit region end(Prohibit modification and deletion)
